### DISCUSSION TITLE
Stuck after submitting passcode to MFA using Google Apps Script

### DISCUSSION THREAD

**Discussion by themonomers** [timdorr/tesla-api#307] [Q&A] [answered]
I get to the /oauth2/v3/authorize/mfa/verify part and the response I get back is:

response: {
  "error": {
    "code": "ERR_EXPIRED",
    "message": "Expired transaction"
  }
}

Everything else seems to work up until that point.  I'm using Google Apps Script.  Any idea on what I'm doing wrong?

**Comment by themonomers** [comment, marked as answer]
Never mind I figured it out.  It's because for the MFA call, it's a JSON payload.

**Reply by nikemeal** [reply]
could you elaborate on this at all?  i'm getting the same and have made sure my header is `content-type: application/json` and that the POSTFIELDS i send are json_encoded yet still get this back.  
is there anything else that specifically has to be done?

**Reply by themonomers** [reply]
Not sure if you know this but none of this works now from Google Apps Scripts.  I had to use Python from my own server to get the token.  Tesla must be blocking the IP's where the Google Scripts are running from.

I still run most of the scripts from Google but just not the token retrieval part.  If you want to know how to do this in Python I can show you.

**Reply by nikemeal** [reply]
Think I figured my own issue out, it looks like when I was passing through the transaction ID i'd json_encoded it for some reason before, so it was now ""12345"" not "12345" which was failing. 
got it to go through now but have got a separate issue with mfa, but at least this one is out the way. 
thanks for the reply though :)
